src/olympia/addons/models.py

```diff
@@ -1713,6 +1713,39 @@ def get_e10s_classname(self):
         return amo.E10S_COMPATIBILITY_CHOICES_API[self.e10s]
 
 
+class AddonApprovalsCounter(ModelBase):
+    """Model holding a counter of the number of times a listed version
+    belonging to an add-on has been approved by a human. Reset everytime a
+    listed version is auto-approved for this add-on."""
+    addon = models.OneToOneField(
+        Addon, primary_key=True, on_delete=models.CASCADE)
+    counter = models.PositiveIntegerField(default=0)
+
+    def __unicode__(self):
+        return u'%s: %d' % (unicode(self.pk), self.counter) if self.pk else u''
+
+    @classmethod
+    def increment_for_addon(cls, addon):
+        """
+        Increment approval counter for the specified addon. If an
+        AddonApprovalsCounter already exists, it updates it, otherwise it
+        creates and saves a new instance.
+        """
+        try:
+            obj = cls.objects.get(addon=addon)
+            obj.update(counter=F('counter') + 1)
+        except cls.DoesNotExist:
+            obj = cls.objects.create(addon=addon, counter=1)
```

Comment:
Yeah, that works, I'll change it. I initially had `cls.objects.update_or_create(addon=addon, defaults={'counter': F('counter') + 1})` but that doesn't work for creations, so I started coping `update_or_create` logic over, but you're right, might as well just `get_or_create()` then `update()`.